### DISCUSSION TITLE
Kiali pod labels don't follow Kubernetes labels recommendations

### DISCUSSION THREAD

**Discussion by unixmind** [kiali/kiali#4879] [General]
![Kiali-label-issue](https://user-images.githubusercontent.com/19151638/161080522-8ab46040-b754-4fed-b9ae-4ece148bd985.jpg)

Kiali expects both "metadata.labels.app" and "metadata.labels.version"

**[Kubernetes recommends "metadata.labels.app.kubernetes.io/name" and "metadata.app.kubernetes.io/version"](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/)**

**Comment by jmazzitelli** [comment]
not a bug - you need to configure Kiali if Istio is not using its default "app" or "version" labels when collecting telemetry. See: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.istio_labels

~Note that you must also tell Istio to use app.kubernetes.io/version (and name) labels - because Istio will be using those labels when generating telemetry and storing in  Prometheus. If you have not done this, configuring Kiali will not be enough for you. Kiali needs to be told what labels **Istio** is using for its telemetry.~ UPDATE: I was reminded this is not true anymore - istio can use either/or sets of labels.

There are discussions about this kind of thing in several github issues/PRs in both istio and kiali repos - I do not have links offhand but google should find them.

**Comment by jshaughn** [comment]
> Note that you must also tell Istio to use app.kubernetes.io/version (and name) labels - because Istio will be using those labels when generating telemetry and storing in Prometheus. 

I don't think this is true.  Istio telemetry sets canonical fields for "app" and "version", and those fields will use the app.kubernetes.io labels, or the istio (app/version) labels, if found.

**Reply by jmazzitelli** [reply]
Ah right - I think they have the ability to use either/or. We talked about this with them. We can't do that because of the performance degredation Kiali would see (due to the fact k8s does not allow you to query based on label name regex). There was discussed over in istio in a github issue a while ago. If I find it, I'll link here.

But the short answer is simple - if you want to use app.kubernetes.io labels, just tell Kiali that - see my earlier link to the docs for where you do that.

**Reply by jmazzitelli** [reply]
Here's what I was thinking of - I'm pretty sure this was discussed in other issues on github but I can't find where.

https://github.com/kiali/kiali/issues/1458#issuecomment-558271458
